civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -2170,6 +2177,121 @@ function civihr_employee_portal_manager_approval_pick_days_callback($ajax, $acti
 }
 
 /**
+ * Function that processes approvals and rejections made by manager from the quick absence approval widget button
+ *
+ * @param int $ajax
+ *   boolean value, 1 if it is an ajax call
+ * @param string $action
+ *   should always be "quick_approval"
+ * @param int $activity_id
+ *   Activity ID that identifies the absence to be processed
+ * @param int $status
+ *   ID of status that should be set to main absence and each absence per day (2: Accepted, 9: Rejected)
+ * @param string $type
+ *   Staus type that is to be set, either "accept" or "reject"
+ */
+function civihr_employee_portal_manager_approval_quick_approval($ajax, $action, $activity_id, $status, $type) {
+  // Civi init
+  civicrm_initialize();
+
+  try {
+    // Load main activity leave
+    $main_activity = civicrm_api3('Activity', 'get', ['sequential' => 1, 'id' => $activity_id])['values'][0];
+
+    // Get activities for each day of leave
+    $activityTypes = CRM_Core_OptionGroup::values('activity_type', FALSE, FALSE, FALSE, NULL, 'name');
```

Comment:
you can just to the OptionValues API to get the activity type where the name = 'Absence' instead of fetching all activity types and then filtering the one you want :

```php
$absenceTypeValue= civicrm_api3('OptionValue', 'get', array(
  'sequential' => 1,
  'return' => array("value"),
  'option_group_id' => "activity_type",
  'name' => "Absence",
  'options' => array('limit' => 1),
))['values'][0];

    $search_params = [
      'source_record_id' => $activity_id,
      'activity_type_id' => CRM_Utils_Array::key('Absence', $absenceTypeValue)
    ];
```